src/zenml/zen_stores/base_zen_store.py

```diff
@@ -317,30 +316,61 @@ def validate_active_config(
         """
         active_project: Optional[ProjectResponse] = None
 
-        if active_project_name_or_id:
+        if active_project_id:
             try:
-                active_project = self.get_project(active_project_name_or_id)
+                active_project = self.get_project(active_project_id)
             except (KeyError, IllegalOperationError):
-                active_project_name_or_id = None
+                active_project_id = None
                 logger.warning(
                     f"The current {config_name} active project is no longer "
                     f"available."
                 )
 
+        if active_project is None:
+            user = self.get_user()
+            if user.default_project_id:
+                try:
+                    active_project = self.get_project(user.default_project_id)
+                except (KeyError, IllegalOperationError):
+                    logger.warning(
+                        "The default project %s for the active user is no longer "
+                        "available.",
+                        user.default_project_id,
+                    )
+                else:
+                    logger.info(
+                        f"Setting the {config_name} active project "
+                        f"to '{active_project.name}'."
+                    )
+
         if active_project is None:
```

Comment:
@stefannica After talking with @htahir1, I reworked this a little:
- If the user does not have a default project set, we try to list all projects
- If there is a single project, we set that as active. This should also cover the OSS case where there should only be a single `default` project.
- If there are no or multiple projects, we don't set any project active. I guess for the multiple ones, I could still check if there is a project called `default` and use that instead, to have an even better fallback for the OSS case?